packages/ckeditor5-image/src/image/imageediting.js

```diff
@@ -134,3 +164,24 @@ export function createImageViewElement( writer ) {
 
 	return figure;
 }
+
+/**
+ * {@link module:engine/view/matcher~Matcher} pattern. Returns function which checks if a given element is `<image>` element that is placed
+ * inside the element of a provided type.
+ *
+ * @param {String} parentType
+ * @returns {Function}
+ */
+function matchImageInsideParent( parentType ) {
+	return element => {
+		const parent = element.parent;
+
+		if ( element.name === 'img' && element.hasAttribute( 'src' ) && // Convert only images with src attribute.
+			parent && parent.name === parentType // Convert only images inside paragraph.
+		) {
+			return { name: true };
+		}
```

Comment:
This way it's much more readable:

```javascript
		// Convert only images with src attribute.
		if ( !element.is( 'element', 'img' ) || !element.hasAttribute( 'src' ) ) {
			return null;
		}

		// Convert only images inside paragraph.
		if ( !parent || !parent.is( 'element', parentType ) {
			return null;
		}

		return { name: true, attributes: [ 'src' ] };
```

Also, this matcher should indicate that the `src` attribute should be consumed: 

`return { name: true, attributes: [ 'src' ] };` 

And what about inline images inside a list item, or in any other element that allows inline images? According to the schema it's allowed everywhere where the text is allowed.

Reply:
And `parentType` is more like `parentName`.

Reply:
> And what about inline images inside a list item, or in any other element that allows inline images? According to the schema it's allowed everywhere where the text is allowed.

Reply:
> 
> 
> > And what about inline images inside a list item, or in any other element that allows inline images? According to the schema it's allowed everywhere where the text is allowed.
This is the topic for the next tasks. In this ticket the only way to create an inline image is to convert it from block image - so it will always be inserted inside a paragraph.

Reply:
But that paragraph can be easily changed by the user to any other kind of block.